include/keisan/matrix/matrix.impl.hpp

```diff
@@ -93,6 +93,24 @@ Matrix<M, N> Matrix<M, N>::zero()
   return matrix;
 }
 
+template<size_t M, size_t N>
+Matrix<N, N> Matrix<M, N>::identity()
+{
+  static_assert(
+    M == N,
+    "The dimensions of matrix are not matched. "
+    "There is no identity matrix for non-square matrix!");
+
+  Matrix<N, N> matrix;
+  for (size_t i = 0; i < N; ++i) {
+    for (size_t j = 0; j < N; ++j) {
+      matrix[i][j] = (i == j) ? 1.0 : 0.0;
+    }
+  }
+
+  return matrix;
+}
```

Comment:
i suggest to return `Matrix<M, N>` instead, even if in the end `M` will always equal to `N`.